src/sfProject.ts

```diff
@@ -602,9 +602,15 @@ export class SfProject {
    */
   public getPackageFromPath(path: string): Optional<NamedPackageDir> {
     const packageDirs = this.getPackageDirectories();
-    const match = packageDirs.find(
-      (packageDir) => basename(path) === packageDir.path || path.includes(packageDir.fullPath)
-    );
+    // resolve the given path
+    const fullPath = resolve(path);
+    const match = packageDirs.find((packageDir) => {
+      // fullPath will not have a trailing slash, so remove it from packageDir.fullPath, if it exists
+      const fullPathSansTrailingSep = packageDir.fullPath.replace(/(\\|\/)$/, '');
+      return (
+        basename(path) === packageDir.path || fullPath === fullPathSansTrailingSep || path.includes(packageDir.fullPath)
+      );
+    });
     return match;
```

Comment:
suggestion (add comments if you want to explain the trailing slash part)
```ts
return this.getPackageDirectories().find(
      (packageDir) =>
        basename(path) === packageDir.path || path.includes(packageDir.fullPath) || packageDir.fullPath.startsWith(resolve(path))
    );
```

Reply:
it puts the `resolve` last so its avoided if possible

Reply:
> it puts the `resolve` last so its avoided if possible

Not following. Are you asking that the use of resolve should be done after the other conditions are checked?

Reply:
If you wanna go with Shane's suggestion make sure to change this:

```diff
-path.includes(packageDir.fullPath)
+packageDir.fullPath.includes(path)

the pkg dir fullPath always have a trailing slash:
https://github.com/forcedotcom/sfdx-core/blob/main/src/sfProject.ts#L246